deps/amqp_client/src/amqp_connection.erl

```diff
@@ -153,12 +157,20 @@ start(AmqpParams) ->
 %% user specified connection name.
 start(AmqpParams, ConnName) when ConnName == undefined; is_binary(ConnName) ->
     ensure_started(),
+    Obfuscate   = fun(Password) -> credentials_obfuscation:encrypt(Password) end,
```

Comment:
You can reference an Erlang function like so: `fun credentials_obfuscation:encrypt/1`, and assign it to a binding:

``` erl
ObfuscateFn = fun credentials_obfuscation:encrypt/1,
```

Reply:
Clever indeed; thank you.